packages/unchained-client/src/cosmossdk/parser/utils.ts

```diff
@@ -24,6 +25,7 @@ const assetIdByDenom = new Map<string, AssetId>([
 
 export const getAssetIdByDenom = (denom: string, assetId: string): AssetId | undefined => {
   if (assetIdByDenom.has(denom)) return assetIdByDenom.get(denom) as AssetId
+  if (assetId === osmosisAssetId) return adapters.osmosisDenomToAssetId(denom)
```

Comment:
this logic essentially lives right below this at L:32. could create the asset id there if we wanted to.

```
if (assetNamespace === 'gamm') {
  return toAssetId({ chainId, assetNamespace: 'ibc', assetReference: denom })
}
```

Reply:
Yeah, at first I thought it might be nice to keep any osmosis-specific parsing stuff out of the cosmossdk base utils, but the doing it here saves several lines and a follow-up PR, so I agree.